### PR TITLE
vulkan: make MMVQ path selection independent of batch size

### PR DESCRIPTION
**Pull request by Ankk98**
## Overview
ggml_vk_should_use_mmvq returned true for any n > 1, so N=1 decode ran F32 DMMV while N>1 speculative verify staged Q8_1 + integer-dot MMVQ. The two precisions disagree, flipping greedy picks at near-ties (#25618).

Drop the batch-size early-out so the vendor/type/k checks decide identically for every N; cases already consistent keep their fast path.
mul_mat_vec_id shares the predicate (existing MUL_MAT_ID suite 883/883
green); prefill GEMM MMQ never consulted it (pp128 identical).
FORCE/DISABLE env overrides unchanged.

Adds MMVQ_BATCH (Q2_0/Q4_0/Q4_1/Q4_K/Q5_0/Q5_1/Q6_K/Q8_0 x F32, N=1 vs N=2 col0 ULP-0): 8/16 fail pre-fix, 16/16 pass post-fix on Vulkan, 16/16 on CPU. Decode timing tg128 43.76 vs 43.67 tok/s default vs FORCE_MMVQ; spec-decode e2e shows no conclusive delta.

## Additional information
**Bug**: Vulkan has two calculators for the same multiply: a precise one and a fast one that rounds the input to 8-bit blocks first. Single-token decode uses the precise one, but as soon as a second token joins the batch (speculative verify checks [sampled, draft] together) it switches to the fast one. The answers differ by up to 0.2, so at any near-tie in the logits - e.g. token 548 at 24.3196 vs token 320 at 24.2953 - the winner flips and the whole continuation diverges from vanilla, even though greedy speculative decoding is supposed to be lossless.

**First principles**: precision is part of the answer's contract, not a runtime tuning knob, so batch size must never select it. Serial N==1 decode is the specification (it's what vanilla runs and goldens pin), and batching is only an optimization that has to reproduce it - so the path decision must be identical for every N. Where MMVQ was already the N==1 path (large-k, NVIDIA Q2/Q3) it stays on both widths; where the widths disagreed they now follow the N==1 decision. Prefill keeps its own fast family since it is never compared token-for-token with decode.

**Validation**: `MMVQ_BATCH` 8/16 fail pre-fix, 16/16 pass post-fix on RADV, 16/16 CPU; `MUL_MAT_ID` 883/883; `tg128` 43.76 vs 43.67 default vs `FORCE_MMVQ`.

```
cmake -B build -DGGML_VULKAN=ON && cmake --build build --target test-backend-ops -j 12
./build/bin/test-backend-ops test -b Vulkan0 -j 8 -o MMVQ_BATCH
```
## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: YES

### CODE REVIEW

**Comment by 0cc4m** [comment]
This needs actual model evidence, not synthetic tests, and even then it is likely not the right solution. It will cost a lot of performance.

**Comment by Ankk98** [comment]
@0cc4m Agreed on performance part, I do have actual model evidence. I will share it soon.
